libr/debug/p/native/linux/linux_debug.c

```diff
@@ -193,52 +206,193 @@ bool linux_set_options (RDebug *dbg, int pid) {
 	return true;
 }
 
-int linux_attach (RDebug *dbg, int pid) {
-	linux_set_options (dbg, pid);
-	int ret = ptrace (PTRACE_ATTACH, pid, NULL, NULL);
+// Local copy for us. Here we keep all traced threads (including the main_pid). 
+static RList *list_traced_threads = NULL;
+
+static int match_pid (const void *pid_o, const void *th_o) {
+	int pid = *(int *)pid_o;
+	RDebug *th = (RDebug *)th_o; 
+	return pid == th->pid;
+}
+
+static void remove_list_thread (void) {
+	RDebugPid *th;
+	RListIter *it;
+	RList *list = list_traced_threads;
+	r_list_foreach (list, it, th) {
+		r_list_delete_data (list, th);
+	}
+	r_list_free (list);
+	eprintf ("remove_list_thread - done\n");
+}
+
+static void add_and_attach_new_thread (RDebug *dbg, int tid) {
+	(void) linux_attach (dbg, tid);
+	eprintf ("add_and_attach_new_thread: buf: %d\n", tid);	
+	r_list_append (list_traced_threads, r_debug_pid_new (NULL, tid, 's', 0));
+}
+
+static int attach_to_ptid (RDebug *dbg, int ptid) {
+	linux_set_options (dbg, ptid);
+	eprintf ("attach_to_ptid: Attaching to pid: %d\n", ptid);
+	return ptrace (PTRACE_ATTACH, ptid, NULL, NULL);
+}
+
+static RList *attach_to_pid_and_threads (RDebug *dbg, int main_pid) {
```

Comment:
When declaring the function and the definition do not use spaces between the name and the parentheses, so it should be `attach_to_pid_and_threads(RDebug *dbg..)`